src/general/OptionsMenu.cs

```diff
@@ -379,142 +461,247 @@ private Viewport.MSAA MSAAIndexToResolution(int index)
         }
     }
 
+    private void CompareSettings()
+    {
+        // Enable the save and reset buttons if the current setting values differ from the saved ones.
+        if (Settings.Instance == savedSettings)
+        {
+            // Settings match
+            resetButton.Disabled = true;
+            saveButton.Disabled = true;
+        }
+        else
+        {
+            // Settings differ
+            resetButton.Disabled = false;
+            saveButton.Disabled = false;
+        }
+    }
+
     /*
-      GUI callbacks
+      GUI Control Callbacks
     */
 
     private void OnBackPressed()
     {
         GUICommon.Instance.PlayButtonPressSound();
 
-        // TODO: only save if something was changed
+        // If any settings have been changed, show a dialogue asking if the changes should be kept or
+        // discarded.
+        if (Settings.Instance != savedSettings)
+        {
+            backConfirmationBox.Visible = true;
```

Comment:
`PopupCenteredMinsize()` should be used so the popup will be shown in modal form, otherwise other buttons outside the dialog box can still be pressed which might lead to funky results.